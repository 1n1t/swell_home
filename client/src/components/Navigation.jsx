/* eslint-disable jsx-a11y/anchor-is-valid,import/no-extraneous-dependencies,react/forbid-prop-types */
import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';

import NavigationLink from './NavigationLink';
import navData from '../data/navigation.json';
import smData from '../data/social-media.json';

class Navigation extends Component {
	static renderSocialMediaLinks() {
  	return smData.map(({url, cssClass}) => (
			<li className="link" key={url}>
				<a
					href={url}
					target="_blank"
					rel="noopener noreferrer"
				>
					<span className={cssClass} />
				</a>
			</li>
		))
	}

	state = {
		overlayColor: ''
	};

	componentDidMount() {
		this.setDefaultOverlayColor();
	}

	setDefaultOverlayColor() {
		const color = this.getDefaultOverlayColor();
		this.setState({overlayColor: color});
	};

	getDefaultOverlayColor = () => navData.filter(({ path }) => path === this.props.location.pathname)[0].overlayColor;

	changeOverlayColorHandler = (color = '') => {
		const overlayColor = (color === '') ? this.getDefaultOverlayColor() : color;
		this.setState({overlayColor});
	};

	renderNavLinks = () => navData.map(item =>
			<NavigationLink
				key={item.path}
				{...item}
				changeOverlayColor={this.changeOverlayColorHandler}
			/>);

	render() {
    const overlayStyle = {
      backgroundColor: this.state.overlayColor
    };

		return (
      <div className="Navigation">
        <nav className="navbar">
          <ul className="links">{this.renderNavLinks()}</ul>
          <ul className="social-media">{Navigation.renderSocialMediaLinks()}</ul>
        </nav>
        <div className="overlay" style={overlayStyle} />
      </div>
    );
  }
}

Navigation.propTypes = {
  location: PropTypes.object.isRequired
};

export default withRouter(Navigation);
