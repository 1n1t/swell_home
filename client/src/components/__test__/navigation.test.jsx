import React from 'react';
import { shallow } from 'enzyme';
import { StaticRouter } from 'react-router-dom';

import Navigation from '../Navigation';

test('the component mounts with defaults', () => {
  const wrapper = shallow(
    <StaticRouter location="/" context={{}}>
      <Navigation />
    </StaticRouter>
  );

  expect(wrapper).toMatchSnapshot();
});
