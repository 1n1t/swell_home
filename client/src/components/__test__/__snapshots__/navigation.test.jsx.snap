// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the component mounts with defaults 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StaticRouter
    basename=""
    context={Object {}}
    location="/"
  >
    <Unknown />
  </StaticRouter>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Unknown />,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Unknown />,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
